cpp/src/gandiva/precompiled/bitmap.cc

```diff
@@ -38,11 +38,7 @@ FORCE_INLINE
 void bitMapSetBit(unsigned char* bmap, int position, bool value) {
   int byteIdx = POS_TO_BYTE_INDEX(position);
   int bitIdx = POS_TO_BIT_INDEX(position);
-  if (value) {
-    bmap[byteIdx] |= (1 << bitIdx);
-  } else {
-    bmap[byteIdx] &= ~(1 << bitIdx);
-  }
+  bmap[byteIdx] ^= (-value ^ bmap[byteIdx]) & (1UL << bitIdx);
```

Comment:
There is some redundancy here with arrow/util/bit-util.h. We should explore merging these together

Reply:
Yeah but AFAIK this code comes under the pre-compiled part of gandiva where we create a LLVM-IR code for the above bitmap operation and use them in the runtime. @pravindra Your insights ? Can we merge both of them ?

Reply:
agree with @wesm. but, it'll require some changes to use this file in the IR generation (eg. we don't want the functions to be inlined at compile time, but later in the optimization pass).

@wesm, Can we use a seperate jira for merging the files ?

Reply:
Yes, separate JIRA is fine